api/bucketsd/client/client_test.go

```diff
@@ -955,6 +957,55 @@ func pushPathAccessRoles(
 	})
 }
 
+func TestClient_AcceptPathAccessRoles(t *testing.T) {
+	ctx, userctx, threadsclient, client := setupForUsers(t)
+	buck, err := client.Create(ctx)
+	require.NoError(t, err)
+
+	q, err := client.PushPaths(ctx, buck.Root.Key)
+	require.NoError(t, err)
+	err = q.AddFile("file", "testdata/file1.jpg")
+	require.NoError(t, err)
+	for q.Next() {
+		require.NoError(t, q.Err())
+	}
+	require.NoError(t, q.Close())
+
+	// Grant Token Access
+	token, tokenHash := generateShareToken(t)
+	err = client.PushPathAccessRoles(ctx, buck.Root.Key, "file", map[string]bucks.Role{
+		tokenHash: bucks.Writer,
+	})
+	require.NoError(t, err)
+
+	// New User Accepts File
+	user2, user2ctx := newUser(t, userctx, threadsclient)
+	role, err := client.AcceptPathAccessRoles(ctx, buck.Root.Key, "file", token, user2.GetPublic().String())
```

Comment:
This is accepting the role _as the bucket creator_, not the user. If you use the `user2ctx`, this will fail because that user doesn't (yet) have access to the path. This is why we need to handle the key swap in the write validator logic: https://github.com/textileio/textile/blob/master/threaddb/buckets.go#L298